src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,29 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH_OF_PASSWORD = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("This login is already taken");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RuntimeException("Sorry! You are younger than " + MIN_AGE);
+        }
+        if (user.getPassword().length() < MIN_LENGTH_OF_PASSWORD) {
+            throw new RuntimeException("Password contains less than " + MIN_LENGTH_OF_PASSWORD);
+        }
+        checkIsNull(user);
```

Comment:
You have to call this method at the beginning, because if user is null you won't get your exception